src/mapping/nav2_map_server/test/unit/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+include_directories( ${PROJECT_SOURCE_DIR}/test )
+
+ament_add_gtest(test_occ_grid test_occ_grid.cpp ${PROJECT_SOURCE_DIR}/test/test_constants.cpp)
+
+ament_target_dependencies(test_occ_grid rclcpp nav_msgs)
+
+target_link_libraries(test_occ_grid
+  map_reps
+  stdc++fs
+  ${ament_LIBRARIES}
```

Comment:
Also, you should not need the ament_LIBRARIES variable at all in this call. The ament_target_dependencies call above sets the target link libraries for ament packages.